src/pybamm_cookiecutter/entry_point.py

```diff
@@ -76,35 +84,35 @@ def get_entries(group_name):
         else:
             return importlib.metadata.entry_points(group=group_name)
 
-    def __new__(cls):
-        """Ensure only one instance of ParameterSets exists"""
-        if not hasattr(cls, "instance"):
+    def __new__(cls, group):
+        """Ensure only two instances of entry points exist, one for parameter sets and the other for models"""
+        if EntryPoint._instances < 2:
```

Comment:
Given that is now <2, is there any chance that someone accidentally initialises 2 instances of parameter sets? As in, would it be safer to check that each can only be initialised once?

Reply:
I think one can add multiple parameter sets (or models) inside one entry point group, so this code is trying to check if someone defines the `[project.entry-points."parameter_sets"]` (or the models counterpart) table more than once in `pyproject.toml`. Is that correct, @santacodes?

Reply:
> Given that is now <2, is there any chance that someone accidentally initialises 2 instances of parameter sets? As in, would it be safer to check that each can only be initialised once?

A user cannot initialise 2 instances of parameter sets accidentally through the imported module itself but they definitely can initialise an instance in `entry_point.py`. In that case, only two instances will be accounted and created, the first two ones mentioned in the file, the third one would simply be the same object of the second one initialised. So any new instances other than the first two will just have the same object address of the second instance like depicted in the screenshot. I'm not sure this explanation was clear enough though so please feel free to mention if I said anything wrong 😅.